catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,706 @@
+        case AttributeDefinition.PASSWORD:
+        case AttributeDefinition.STRING:
+        default:
+            return value;
+        }
+    }
+
+    protected BundleContext getBundleContext() {
+        return FrameworkUtil.getBundle(this.getClass())
+                .getBundleContext();
+    }
+
+    private void updateRegistrySourceConfigurations() {
+        try {
+            List<Metacard> metacards = federationAdminService.getRegistryMetacards();
+
+            for (Metacard metacard : metacards) {
+                try {
+                    updateRegistryConfigurations(metacard, false);
+                } catch (InvalidSyntaxException | ParserException | IOException e) {
+                    LOGGER.error(
+                            "Unable to update registry configurations. Registry source configurations won't be updated for metacard id: {}",
+                            metacard.getId());
+                }
+            }
+
+        } catch (FederationAdminException e) {
+            LOGGER.error(
+                    "Error getting registry metacards. Registry source configurations won't be updated.");
+        }
+    }
+
+    public void setActivateConfigurations(boolean activateConfigurations) {
+        if (!(activateConfigurations == this.activateConfigurations)) {
+            this.activateConfigurations = activateConfigurations;
+
+            if (!activateConfigurations || preserveActiveConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setPreserveActiveConfigurations(boolean preserveActiveConfigurations) {
+        if (!(preserveActiveConfigurations == this.preserveActiveConfigurations)) {
+            this.preserveActiveConfigurations = preserveActiveConfigurations;
+
+            if (preserveActiveConfigurations || !activateConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setCleanUpOnDelete(boolean cleanUpOnDelete) {
+        this.cleanUpOnDelete = cleanUpOnDelete;
+    }
+
+    public void setBindingTypeFactoryPid(List<String> bindingTypeFactoryPid) {
+        bindingTypeToFactoryPidMap.clear();
+
+        for (String mapping : bindingTypeFactoryPid) {
+            String bindingType = StringUtils.substringBefore(mapping, "=");
+            String factoryPid = StringUtils.substringAfter(mapping, "=");
+
+            if (StringUtils.isNotBlank(bindingType) && StringUtils.isNotBlank(factoryPid)) {
+                bindingTypeToFactoryPidMap.put(bindingType, factoryPid);
+            }
+        }
+    }
+
+    public void setSourceActivationPriorityOrder(List<String> sourceActivationPriorityOrder) {
+        if (sourceActivationPriorityOrder != null
+                && !sourceActivationPriorityOrder.equals(this.sourceActivationPriorityOrder)) {
+            this.sourceActivationPriorityOrder = sourceActivationPriorityOrder;
```

Comment:
This is a `CopyOnWriteArrayList()` so we probably don't just want to replace it with the one provided.